src/pages/iou/MoneyRequestModal.js

```diff
@@ -0,0 +1,499 @@
+        const amountInDollars = Math.round(amount * 100);
+        const currency = props.iou.selectedCurrencyCode;
+        const trimmedComment = comment.trim();
+        const participant = participants[0];
+
+        if (paymentMethodType === CONST.IOU.PAYMENT_TYPE.ELSEWHERE) {
+            IOU.sendMoneyElsewhere(
+                props.report,
+                amountInDollars,
+                currency,
+                trimmedComment,
+                props.currentUserPersonalDetails.login,
+                participant,
+            );
+            return;
+        }
+
+        if (paymentMethodType === CONST.IOU.PAYMENT_TYPE.PAYPAL_ME) {
+            IOU.sendMoneyViaPaypal(
+                props.report,
+                amountInDollars,
+                currency,
+                trimmedComment,
+                props.currentUserPersonalDetails.login,
+                participant,
+            );
+            return;
+        }
+
+        if (paymentMethodType === CONST.IOU.PAYMENT_TYPE.EXPENSIFY) {
+            IOU.sendMoneyWithWallet(
+                props.report,
+                amountInDollars,
+                currency,
+                trimmedComment,
+                props.currentUserPersonalDetails.login,
+                participant,
+            );
+        }
+    }
+
+    /**
+     * @param {Array} selectedParticipants
+     */
+    function createTransaction(selectedParticipants) {
+        const reportID = lodashGet(props, 'route.params.reportID', '');
+        const trimmedComment = comment.trim();
+
+        // IOUs created from a group report will have a reportID param in the route.
+        // Since the user is already viewing the report, we don't need to navigate them to the report
+        if (props.hasMultipleParticipants && CONST.REGEX.NUMBER.test(reportID)) {
+            IOU.splitBill(
+                selectedParticipants,
+                props.currentUserPersonalDetails.login,
+                amount,
+                trimmedComment,
+                props.iou.selectedCurrencyCode,
+                props.preferredLocale,
+                reportID,
+            );
+            return;
+        }
+
+        // If the IOU is created from the global create menu, we also navigate the user to the group report
+        if (props.hasMultipleParticipants) {
+            IOU.splitBillAndOpenReport(
+                selectedParticipants,
+                props.currentUserPersonalDetails.login,
+                amount,
+                trimmedComment,
+                props.iou.selectedCurrencyCode,
+                props.preferredLocale,
+            );
+            return;
+        }
+        IOU.requestMoney(
+            props.report,
+            Math.round(amount * 100),
+            props.iou.selectedCurrencyCode,
+            props.currentUserPersonalDetails.login,
+            selectedParticipants[0],
+            trimmedComment,
+        );
+    }
+
+    function renderHeader() {
```

Comment:
Thoughts on just creating a new "Header" component we can use for this? This feels a little "heavy" as an inline function and only has a few dependencies that can be passed as props.

Reply:
In one of my commits, I started breaking this out into a separate header - I ended up moving it back since I had to pass a bunch of props that were very specific to this file and felt like it could be better generalized in a separate PR that refactors it

Reply:
Added this issue for cleanup later https://github.com/Expensify/App/issues/16890

Reply:
Actually, it's not that bad so will just do it here